sdk/include/xdk/haltypes.h

```diff
@@ -1,5 +1,13 @@
 /* Hardware Abstraction Layer Types */
 
+$if (_WDMDDK_)
+typedef
+NTSTATUS
+NTAPI
+PROCESSOR_HALT_ROUTINE(
+    _Inout_opt_ PVOID Context
+);
+$endif (_WDMDDK_)
```

Comment:
```suggestion

typedef PROCESSOR_HALT_ROUTINE *PPROCESSOR_HALT_ROUTINE;
$endif (_WDMDDK_)

```
The `PPROCESSOR_HALT_ROUTINE` is also present in the wdm.h